### ISSUE TITLE
pygvt looks for mpicxx library but OpenMPI provides mpi_cxx

### ISSUE THREAD

**Issue by ngoldbaum** [TACC/GraviT#10]
`pygvt` fails to build on my setup with the following error:

```
clang++ -bundle -undefined dynamic_lookup build/temp.macosx-10.12-x86_64-3.6/src/gvt/gvt.o -L/Users/goldbaum/Documents/embree-2.17.0.x86_64.macosx/lib -L/usr/local/lib/lib64 -L/usr/local/lib -L/usr/local/lib/lib -L/Users/goldbaum/Documents/iceT/lib -lgvtRender -lgvtCore -lIceTGL -lIceTMPI -lIceTCore -lembree -lboost_system-mt -lmpi -lmpicxx -o /Users/goldbaum/Documents/GraviT/pygvt/gvt.cpython-36m-darwin.so -std=c++11 -mmacosx-version-min=10.11 -stdlib=libc++
ld: warning: directory not found for option '-L/usr/local/lib/lib64'
ld: warning: directory not found for option '-L/usr/local/lib/lib'
ld: library not found for -lmpicxx
```

I was able to fix this locally by replacing `mpicxx` in the `pygvt` `setup.py` file with `mpi_cxx`.

**Comment by pnav** [comment]
Thanks for the detailed report, Nathan! We should have this look for both versions. It would be awesome if MPI implementations could agree on a single library syntax :-)

Paul

On Oct 18, 2017, at 2:22 PM, Nathan Goldbaum <notifications@github.com<mailto:notifications@github.com>> wrote:


pygvt fails to build on my setup with the following error:

clang++ -bundle -undefined dynamic_lookup build/temp.macosx-10.12-x86_64-3.6/src/gvt/gvt.o -L/Users/goldbaum/Documents/embree-2.17.0.x86_64.macosx/lib -L/usr/local/lib/lib64 -L/usr/local/lib -L/usr/local/lib/lib -L/Users/goldbaum/Documents/iceT/lib -lgvtRender -lgvtCore -lIceTGL -lIceTMPI -lIceTCore -lembree -lboost_system-mt -lmpi -lmpicxx -o /Users/goldbaum/Documents/GraviT/pygvt/gvt.cpython-36m-darwin.so -std=c++11 -mmacosx-version-min=10.11 -stdlib=libc++
ld: warning: directory not found for option '-L/usr/local/lib/lib64'
ld: warning: directory not found for option '-L/usr/local/lib/lib'
ld: library not found for -lmpicxx


I was able to fix this locally by replacing mpicxx in the pygvt setup.py file with mpi_cxx.

**Comment by jgbarbosa** [comment]
Thanks Nathan.

I will try to see if there a better way to detect the dependencies for the python to avoid all these diferences in name convention between linux and mac. So I will let this issue stay open.

The fix on the branch is temporary. A more robust fix should be available in the next couple of days.


On Oct 18, 2017, at 2:24 PM, Paul Navrátil <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks for the detailed report, Nathan! We should have this look for both versions. It would be awesome if MPI implementations could agree on a single library syntax :-)

Paul

On Oct 18, 2017, at 2:22 PM, Nathan Goldbaum <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


pygvt fails to build on my setup with the following error:

clang++ -bundle -undefined dynamic_lookup build/temp.macosx-10.12-x86_64-3.6/src/gvt/gvt.o -L/Users/goldbaum/Documents/embree-2.17.0.x86_64.macosx/lib -L/usr/local/lib/lib64 -L/usr/local/lib -L/usr/local/lib/lib -L/Users/goldbaum/Documents/iceT/lib -lgvtRender -lgvtCore -lIceTGL -lIceTMPI -lIceTCore -lembree -lboost_system-mt -lmpi -lmpicxx -o /Users/goldbaum/Documents/GraviT/pygvt/gvt.cpython-36m-darwin.so -std=c++11 -mmacosx-version-min=10.11 -stdlib=libc++
ld: warning: directory not found for option '-L/usr/local/lib/lib64'
ld: warning: directory not found for option '-L/usr/local/lib/lib'
ld: library not found for -lmpicxx


I was able to fix this locally by replacing mpicxx in the pygvt setup.py file with mpi_cxx.

**Comment by ngoldbaum** [comment]
Fixed by #16.
